### ISSUE TITLE
Artboards fused

### ISSUE THREAD

**Issue by zwei7** [hsw107/Photoshop-Export-Layers-to-Files-Fast#135]
I don't think this script has multiple artboard support.

I have 4 Artboards (6000x4000 pixels) each with 22 layers (88 layers total) and there is transparency.
When I run the script it outputs 88 pngs, but each one has the incorrect canvas size because it fuses all 4 artboards together (12000x8000). The output file has the layer I want, but it is floating in a massively blank empty canvas.

Ideally each png is the size of the artboard it is placed on.

Thanks!

**Comment by ANovitsky** [comment]
Yes, I have the same situation. @zwei7, Did you find any solution?

**Comment by zwei7** [comment]
I deleted 3 artboards, keeping only 1 remaining
I ran the script with 1 artboard (22 layers).
Undid my 3 artboard deletion and did this 3 more times till all 4 artboards were saved individually.
It is a bad work around if you have 1 layer per artboard though.
Sortoff a really bad work around.

**Comment by damat** [comment]
Same issue but even worse: I have 4 boards for 4 sizes of icons.  If the script could go per each Artboard, it would be just perfect and superior compared to any other option. But now there's almost no difference with the default export =((

**Comment by hsw107** [comment]
Unfortunately most of this script was written before Photoshop had artboard support, so it is not something that's built in at the moment at all.
I can do some research and see what it might take, though off-hand it sounds really complicated to integrate, but I totally understand the use-cases here. Sorry for the poor workaround.

**Comment by zwei7** [comment]
Thanks for making this plug-in hsw107

**Comment by inspiredearth** [comment]
Thank you @hsw107 Hanna for your work on this script.
If you ever find the time and inclination, I'd also love to see artboard support in your script.
